quill-jdbc/src/main/scala/io/getquill/sources/jdbc/ConfigJdbcSource.scala

```diff
@@ -0,0 +1,17 @@
+package io.getquill.sources.jdbc
+
+import java.sql.Connection
+
+import io.getquill.JdbcSourceConfig
+import io.getquill.naming.NamingStrategy
+import io.getquill.sources.sql.idiom.SqlIdiom
+
+import scala.util.DynamicVariable
+
+class ConfigJdbcSource[D <: SqlIdiom, N <: NamingStrategy](config: JdbcSourceConfig[D, N])
+  extends JdbcSource[D, N](config) {
+
+  private val currentConnection = new DynamicVariable[Option[Connection]](Some(dataSource.getConnection))
```

Comment:
Removing `DynamicVariable` causes a timeout exception to most of the related tests:

```
Exception encountered when attempting to run a suite with class name: io.getquill.sources.jdbc.mysql.DepartmentsJdbcSpec *** ABORTED ***
   java.sql.SQLTransientConnectionException: HikariPool-2 - Connection is not available, request timed out after 30000ms.
   at com.zaxxer.hikari.pool.HikariPool.getConnection(HikariPool.java:196)
   at com.zaxxer.hikari.pool.HikariPool.getConnection(HikariPool.java:147)
   at com.zaxxer.hikari.HikariDataSource.getConnection(HikariDataSource.java:83)
   at io.getquill.sources.jdbc.ConfigJdbcSource.connection(ConfigJdbcSource.scala:14)
   at io.getquill.sources.jdbc.JdbcSource.transaction(JdbcSource.scala:44)
   at io.getquill.sources.jdbc.mysql.DepartmentsJdbcSpec.beforeAll(DepartmentsJdbcSpec.scala:9)
   at org.scalatest.BeforeAndAfterAll$class.beforeAll(BeforeAndAfterAll.scala:187)
   at io.getquill.sources.jdbc.mysql.DepartmentsJdbcSpec.beforeAll(DepartmentsJdbcSpec.scala:6)
   at org.scalatest.BeforeAndAfterAll$class.run(BeforeAndAfterAll.scala:253)
   at io.getquill.sources.jdbc.mysql.DepartmentsJdbcSpec.run(DepartmentsJdbcSpec.scala:6)
```

Any suggestions?